pkg/windows/calico.go

```diff
@@ -0,0 +1,549 @@
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range secrets.Items {
+		if !isCalicoNodeToken(&secret) {
+			continue
+		}
+		value, ok := secret.Data["token"]
+		if !ok {
+			continue
+		}
+		calicoKubeConfig.Token = string(value)
+	}
+	if calicoKubeConfig.Token == "" {
+		return nil, errors.New("could not retrieve calico node token from the cluster")
+	}
+
+	cfg := &CniConfig{NodeConfig: nodeConfig, NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+	if err := getDefaultConfig(cfg, nodeConfig); err != nil {
+		return nil, err
+	}
+
+	cfg.CalicoConfig.KubeConfig = calicoKubeConfig
+	if err := getCniConfigOverrides(cfg, restConfig); err != nil {
+		return nil, err
+	}
+
+	// TODO: Fix this to the package
+	//agent.SetHnsNetworkName(CalicoHnsNetworkName)
+
+	if err := createNodeNameFile(cfg); err != nil {
+		return nil, err
+	}
+	if err := createKubeConfig(cfg); err != nil {
+		return nil, err
+	}
+
+	if err := createCniConfig(cfg); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// Start starts the CNI services on the Windows node.
+func (c *Calico) Start(config *CniConfig) error {
+	if err := generateCalicoNetworks(config.CalicoConfig.NetworkingBackend); err != nil {
+		return err
+	}
+
+	if err := startCalico(config.CalicoConfig); err != nil {
+		return err
+	}
+	// TODO: Test without it
+	time.Sleep(5 * time.Second)
+
+	if err := startFelix(config.CalicoConfig); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNodeNameFile creates the node name file required by the CNI.
+func createNodeNameFile(config *CniConfig) error {
+	return util2.WriteFile(filepath.Join(GetWindowsAgentDir(), CalicoNodeNameFileName), config.NodeConfig.AgentConfig.NodeName)
+}
+
+// createKubeConfig creates the kube config required by the CNI.
+func createKubeConfig(config *CniConfig) error {
+	kubeTemplate, err := ParseTemplateFromConfig(CalicoKubeConfigTemplate, config)
+	if err != nil {
+		return err
+	}
+	return util2.WriteFile(filepath.Join(GetWindowsAgentDir(), CalicoKubeConfigName), kubeTemplate)
+}
+
+// createCniConfig creates the CNI config for the CNI.
+func createCniConfig(config *CniConfig) error {
+	parsedTemplate, err := ParseTemplateFromConfig(CalicoConfigTemplate, config)
+	if err != nil {
+		return err
+	}
+	return util2.WriteFile(filepath.Join(config.NodeConfig.AgentConfig.CNIConfDir, CalicoConfigName), parsedTemplate)
+}
+
+// GetDefaultConfig sets the default configuration.
+func getDefaultConfig(config *CniConfig, nodeConfig *config.Node) error {
+	platformType, err := GetPlatformType()
+	if err != nil {
+		return err
+	}
+
+	calicoCfg := &CalicoConfig{
+		Name:                  "Calico",
+		Hostname:              nodeConfig.AgentConfig.NodeName,
+		NodeNameFile:          filepath.Join(GetWindowsAgentDir(), "calico_node_name"),
+		KubeNetwork:           "Calico.*",
+		Mode:                  "vxlan",
+		ServiceCidr:           "10.42.0.0/16",
```

Comment:
Is it possible to source these default values from other parts of RKE2? Rather than statically defining them here

Reply:
If there is a way to source these that would be great.